src/backend/access/heap/heapam.c

```diff
@@ -396,9 +397,23 @@ heapgetpage(TableScanDesc sscan, BlockNumber page)
 	 */
 	CHECK_FOR_INTERRUPTS();
 
+	/* Prefetch next block */
+	if (enable_seqscan_prefetch)
+	{
+		int prefetch_limit = seqscan_prefetch_buffers;
+		ParallelBlockTableScanWorker pbscanwork = scan->rs_parallelworkerdata;
+		if (pbscanwork != NULL && pbscanwork->phsw_chunk_remaining < prefetch_limit)
+			prefetch_limit = pbscanwork->phsw_chunk_remaining;
+		if (page + prefetch_limit >= scan->rs_nblocks)
+			prefetch_limit = scan->rs_nblocks - page - 1;
+		for (int i = 1; i <= prefetch_limit; i++)
+			PrefetchBuffer(scan->rs_base.rs_rd, MAIN_FORKNUM, page+i);
+	}
```

Comment:
Does the current heap infrastructure not prefetch their buffers?

Additionally, this should probably be gated behind `#ifdef USE_PREFETCH`.